json-endpoints/jetpack/json-api-jetpack-endpoints.php

```diff
@@ -697,6 +697,27 @@
 	'example_request' => 'https://public-api.wordpress.com/rest/v1.1/sites/example.wordpress.org/sync/settings'
 ) );
 
+// GET /sites/%s/sync-object
+new Jetpack_JSON_API_Sync_Object( array(
+	'description'     => 'Get an object by ID from one of the sync modules, in the format it would be synced in',
+	'group'           => '__do_not_document',
+	'method'          => 'GET',
+	'path'            => '/sites/%s/sync-object',
```

Comment:
Maybe we should put it under `GET /sites/%s/sync/object` in order to have all the `sync` related endpoints under the same path.